drivers/at_cmd/at_cmd.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <logging/log.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <net/socket.h>
+#include <init.h>
+#include <bsd_limits.h>
+
+#include <at_cmd.h>
+
+LOG_MODULE_REGISTER(at_cmd, CONFIG_AT_CMD_LOG_LEVEL);
+
+#define RESPONSE_MAX_LEN  CONFIG_AT_CMD_RESPONSE_MAX_LEN
+#define THREAD_PRIORITY   K_PRIO_PREEMPT(CONFIG_AT_CMD_THREAD_PRIO)
+
+#define AT_CMD_OK_STR    "OK"
+#define AT_CMD_ERROR_STR "ERROR"
+#define AT_CMD_CMS_STR   "+CMS:"
+#define AT_CMD_CME_STR   "+CME:"
+
+static K_THREAD_STACK_DEFINE(socket_thread_stack, 1024);
```

Comment:
If the user changes `RESPONSE_MAX_LEN`, you will need a bigger stack. Are you sure this value should be hardcoded?

Reply:
What is a good expression for this value or should I make it into a KConfig parameter?

Reply:
`char buf[RESPONSE_MAX_LEN];` is a local Thread variable so IMO the stack value should depends on this KConfig. Don't you think so?

Reply:
Yes, and bytes_read, return_code and write_with_response, so my expression would look something like this

#define THREAD_STACK_SIZE RESPONSE_MAX_LEN + 4 + 4 + 4 + ?

The thread also have some metadata, is the stack also used for that or is that put on the global stack? And, I call some functions, is this stack also use for call stack?

I think the documentation is missing information to help me make an educated guestimate here. So for me the value ends up as something that is large enough to fit my use-cases. That is why I proposed to just make it into a KConfig parameter so that user can do their own tuning. But I agree that it's the user needs be able to tune this value in some way.

Reply:
`RESPONSE_MAX_LEN ` + some margin is ok I think. By doing so the program will not crash if the customer increase `RESPONSE_MAX_LEN`.